pkg/backend/identityapi/register_client.go

```diff
@@ -13,16 +14,25 @@ import (
 	"strings"
 	"time"
 
+	"github.com/antihax/optional"
 	backendhttp "github.com/newrelic/infrastructure-agent/pkg/backend/http"
 	"github.com/newrelic/infrastructure-agent/pkg/backend/inventoryapi"
 	"github.com/newrelic/infrastructure-agent/pkg/entity"
+	"github.com/newrelic/infrastructure-agent/pkg/identity-client"
+	"github.com/newrelic/infrastructure-agent/pkg/integrations/v4/protocol"
 	"github.com/newrelic/infrastructure-agent/pkg/log"
 )
 
-var rlog = log.WithComponent("IdentityRegisterClient")
+var rlog = log.WithComponent("identityapi.RegisterClient")
 
-type IdentityRegisterClient interface {
-	Register(agentEntityID entity.ID, entities []RegisterEntity) ([]RegisterEntityResponse, time.Duration, error)
+const (
+	identityPath = "/identity/v1"
+)
+
+type RegisterClient interface {
```

Comment:
then I guess we should rename `NewIdentityRegisterClient` as `NewRegisterClient`

Reply:
Yep, I wasn't very consistent with my naming (or renaming). Shall make the change 👍